src/Azure.Functions.Cli/Actions/LocalActions/InitAction.cs

```diff
@@ -404,6 +404,15 @@ private static async Task WriteLocalSettingsJson(WorkerRuntime workerRuntime, Pr
 
         private static async Task WriteDockerfile(WorkerRuntime workerRuntime, string language, string targetFramework, bool csx)
         {
+            if (WorkerRuntimeLanguageHelper.IsDotnet(workerRuntime))
+            {
+                var functionAppRoot = ScriptHostHelpers.GetFunctionAppRootDirectory(Environment.CurrentDirectory);
+                string projectFilePath = ProjectHelpers.FindProjectFile(functionAppRoot);
+                if (projectFilePath != null)
+                {
+                    targetFramework = await DotnetHelpers.DetermineTargetFramework(Path.GetDirectoryName(projectFilePath));
```

Comment:
Can you check this works when both solution file and project file are present in the same directory? We may need to use the project file instead of the directory when running the `dotnet build -getproperty:TargetFramework` (inside the `DetermineTargetFramework` method)

Reply:
Thanks @kshyju for the help. will do the required test and let you know

Reply:
> it will work if we run the command "dotnet build extension.csproj -getproperty:TargetFramework". Instead of making changes in Dotnet Helper method (DetermineTargetFramework ) I came up with workaround to get Projectroot Element with the help of project file path and get the value of target framework

Is there any reason to not use `DetermineTargetFramework` (with the slight change it needs to support this use case)?  PR #3604 added that method to address some other issues (see the PR for details) and I think it is best if we use that method. We may need to tweak that method to handle this scenario.